kernel/src/osquery.cpp

```diff
@@ -94,6 +159,15 @@ static int subscribe_to_event(osquery_event_t event, int subscribe) {
   }
 
   switch (event) {
+  case OSQUERY_PROCESS_EVENT:
+    if (subscribe && osquery.fileop_listener == NULL) {
+      osquery.fileop_listener =
+          kauth_listen_scope(KAUTH_SCOPE_FILEOP, fileop_scope_callback, NULL);
+    } else if (osquery.fileop_listener) {
```

Comment:
Should this path execute if:
```cpp
subscribe = 1;
osquery.fileop_listener = 0xABCDE000;
```

Or, should we check that `subscribe == 0` too?